packages/suite/src/hooks/wallet/useSendForm.ts

```diff
@@ -54,8 +56,36 @@ const getStateFromProps = (props: UseSendFormProps) => {
         value: props.localCurrency,
         label: props.localCurrency.toUpperCase(),
     };
+
+    // exclude utxos from default composeTransaction process (see sendFormBitcoinActions)
+    // utxos are stored as dictionary where:
+    // `key` is an outpoint (string combination of utxo.txid + utxo.vout)
+    // `value` is the reason
+    // utxos might be spent using CoinControl feature
+    const excludedUtxos: UseSendFormState['excludedUtxos'] = {};
+    if (account.utxo) {
+        const targetAnonymity = props.coinjoinAccount?.targetAnonymity || 0;
+        const anonymitySet = account.addresses?.anonymitySet || {};
+        account.utxo?.forEach(utxo => {
+            const outpoint = utxo.txid + utxo.vout.toString();
+            const anonymity = anonymitySet[utxo.address] || 0;
+            if (anonymity < targetAnonymity) {
+                // didn't reach desired anonymity (coinjoin account)
+                excludedUtxos[outpoint] = 'anonymity';
+            } else if (new BigNumber(utxo.amount).lt(Number(coinFees.dustLimit))) {
```

Comment:
those are excluded also in Connect, right? And we should keep the logic there for other apps but it doesn't hurt to have it here, it's just redundant. Am I getting this right, @szymonlesisz ?

Reply:
yes they are excluded in connect, but this way im grouping them in to `excludedUtxo` and they should be used in coin control in group "dust"